anon/elgam_test.go

```diff
@@ -0,0 +1,141 @@
+package anon
+
+import (
+	"fmt"
+	"testing"
+	"encoding/hex"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/openssl"
+)
+
+// testing ElGam via simplified ExampleSign_1 in sig.go
+func TestElGam( t *testing.T) {
+	// Crypto setup
+	suite := openssl.NewAES128SHA256P256()
+	rand := abstract.HashStream(suite, []byte("example"), nil)
+
+	// Create a public/private keypair (X,x)
+	x := suite.Secret().Pick(rand)		// create a private key x
+	X := suite.Point().Mul(nil,x)		// corresponding public key X
+
+	// Generate the signature
+	M := []byte("Hello World!")		// message we want to sign
+	sig := SignElGam(suite, rand, M, X, x)
+	fmt.Print("Signature:\n"+hex.Dump(sig))
+
+	// Verify the signature against the correct message
+	err := VerifyElGam(suite, M, X, sig)
+	if err != nil {
+		t.Error(err.Error())
+		// panic(err.Error())
+	}
+	fmt.Println("Signature verified against correct message.")
+
+	// Verify the signature against the wrong message
+	BAD := []byte("Goodbye world!")
+	err = VerifyElGam(suite, BAD, X, sig)
+	if err == nil {
+		t.Error("Signature verified against wrong message!?")
+		// panic("Signature verified against wrong message!?")
+	}
+	fmt.Println("Verifying against wrong message: "+err.Error())
+
+	// Output:
+	// Signature:
+	// 00000000  aa ed 0d ac 14 0d 60 d9  88 8b 3c c2 86 0b f5 79  |......`...<....y|
+	// 00000010  bd 1e ec 0c e8 03 83 44  9b 3b 64 c5 14 4e 87 c4  |.......D.;d..N..|
+	// 00000020  58 8c a0 be c8 4c 15 c9  83 56 70 60 23 7a bd 2a  |X....L...Vp`#z.*|
+	// 00000030  1b 64 f1 71 09 ea 0f b3  0d 3d 05 a7 c0 95 c0 94  |.d.q.....=......|
+	// Signature verified against correct message.
+	// Verifying against wrong message: invalid signature
+}
+
+// testing ElGam, ElGam  generated signature, sig.go verified signature
+func TestElGamAgainstSig1( t *testing.T) {
```

Comment:
Good, but since this is just an example code, just one small test is probably adequate - again see crypto/enc_test.go.  Simplicity and understandability are more important than exhaustiveness here.